### PR TITLE
Refactor upload image code, resolves #260

### LINKED ISSUE #260: Frontend - Refactor upload image code

### PR DESCRIPTION
**Pull request by danielcj2**
On a second thought, I'll be leaving this pull request up and keep pushing to it as I tweak the features. Will request reviews once everything is completed.

### CODE REVIEW

File: Client/src/Components/Avatar/index.jsx [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import { Avatar as MuiAvatar } from "@mui/material";
 import PropTypes from "prop-types";
+import { useSelector } from "react-redux";
+import { useEffect, useState } from "react";
+import { bufferTo64, bufferToUrl } from "../../Utils/fileUtils";
```
Comment (ajhollid) [inline Client/src/Components/Avatar/index.jsx:5]: This is unused now, we can remove this

File: Client/src/Utils/fileUtils.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  });
+};
+
+export const bufferToUrl = (buffer) => {
+  console.log(buffer);
+  const blob = new Blob([buffer.data], { type: buffer.contentType });
+  const url = URL.createObjectURL(blob);
+
+  return url;
+};
```
Comment (ajhollid) [inline Client/src/Utils/fileUtils.js:31]: I don't think we need these buffer conversion functions anymore do we?

File: Client/src/Components/Avatar/index.jsx [hunk 3]
```diff
@@ -15,29 +16,48 @@ import PropTypes from "prop-types";
...
+      setImage(`data:image/png;base64,${user.avatarImage}`);
+    }
+  }, [user?.avatarImage]);
+
   return (
     <MuiAvatar
-      alt={`${firstName} ${lastName}`}
-      src={src}
+      alt={`${user?.firstname} ${user?.lastname}`}
+      children={`${user.firstname?.charAt(0)}${user.lastname?.charAt(0)}`}
```
Comment (ajhollid) [inline Client/src/Components/Avatar/index.jsx:35]: React doesn't like this:

`Do not pass children as props. Instead, nest children between the opening and closing tags.eslint[react/no-children-prop](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/no-children-prop.md)`

Any reason they can't be passed regularly inbetween the tags?

**Review by ajhollid — CHANGES_REQUESTED** [review]
Looks good, just some unused imports and a react warning

File: Client/src/Components/Avatar/index.jsx (hunk 3 above)
Comment (danielcj2) [reply Client/src/Components/Avatar/index.jsx:35]: No reason in particular, I was following the @mui docs. Will pass the initials inside the tags.

**Review by ajhollid — APPROVED** [review]
Awesome, thank you!

**Pull request merged**